db/virtual_tables.cc

```diff
@@ -113,6 +113,7 @@ class token_ring_table : public streaming_virtual_table {
         auto id = generate_legacy_id(system_keyspace::NAME, "token_ring");
         return schema_builder(system_keyspace::NAME, "token_ring", std::make_optional(id))
             .with_column("keyspace_name", utf8_type, column_kind::partition_key)
+            .with_column("table_name", utf8_type, column_kind::clustering_key)
```

Comment:
Adding a key column shouldn't be done lightly. But I guess no one knows about this table.

Reply:
Note if someone pages through this table during a rolling upgrade, they'll get a nasty surprise.

Reply:
How? All pages have to be served by the same node. The table has local replication strategy. It is the same as reading a local system table.

Reply:
> Adding a key column shouldn't be done lightly. But I guess no one knows about this table.

I agree. But I think, in this case, any pre-existing query someone might have, will keep working.

Reply:
I agree the request will be the same - "SELECT ... WHERE keyspace=..." was a whole-partition read before, and continues to be. But the result of such a query can be different and can confuse an application, which expected to get the ring for one table (by asking about its keyspace) and suddenly gets a whole array of rings for all tables. But an existing application with existing data without tablets won't even have this problem - it will get back just one entry with the "ALL" table name (if I understood correctly).

Seeing that the only place this API is documented is a developer-only document docs/dev/system_keyspace.md, I tend to guess that nobody is using this API, but I don't really know.